src/json.rs

```diff
@@ -145,7 +146,7 @@ pub mod prelude {
         to_vec_pretty,
     };
     #[cfg(feature = "simd-json")]
```

Comment:
I believe the feature flag name here should be the one with the underscore:
```suggestion
    #[cfg(feature = "simd_json")]
```

Otherwise, only `simd-json` gets pulled in.

Reply:
Ah good catch. That's actually fine though (it was in the code before this PR, even) - the `simd-json` feature flag is an automatic feature from Cargo that cannot be turned off that's always set if simd-json was pulled in. The `simd_json` is our own feature flag that aliases to `simd-json` for naming consistency. So gating by `simd-json` or `simd_json` does the same..

It is confusing though, I give you that.

Should I change all places in the codebase that gate by `simd-json` to say `simd_json`?

Reply:
> Should I change all places in the codebase that gate by `simd-json` to say `simd_json`?

We should be using our feature flag, i.e. `simd_json`. So yes, please. Not just in code, but also [in the test CI job](https://github.com/serenity-rs/serenity/blob/next/.github/workflows/ci.yml#L43), which is why it currently fails.

Reply:
Done